benchmarks/src/main/kotlin/bfschannel/BfsChannelBenchmark.kt

```diff
@@ -0,0 +1,320 @@
+@file:JvmName("BfsChannelBenchmark")
+
+package bfschannel
+
+
+import kotlinx.coroutines.*
+import kotlinx.coroutines.channels.*
+import java.io.FileInputStream
+import java.io.FileOutputStream
+import java.io.InputStreamReader
+import java.io.PrintWriter
+import java.net.URL
+import java.nio.channels.Channels
+import java.nio.file.Paths
+import java.util.*
+import java.util.concurrent.ConcurrentSkipListSet
+import java.util.concurrent.atomic.AtomicLong
+import java.util.zip.GZIPInputStream
+import kotlin.collections.ArrayList
+import kotlin.math.pow
+import kotlin.math.sqrt
+
+private val GRAPH_FILES = listOf(
+        Triple("RAND-1M-10M", "rand", "1000000 10000000"), // 1M nodes and 10M edges
```

Comment:
Please, create a sealed class for graph descriptions. Each instance knows whether it should download or generate a graph and how to parse it.